platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java

```diff
@@ -274,6 +274,21 @@ public Response processSamlResponse(org.opensaml.saml.saml2.core.Response samlRe
                         relayUri.getQuery(),
                         relayUri.getFragment());
             }
+            //this avoids the user ever being redirected to the built in login page if they've already
+            //logged in via the IDP.
+            if ((relayUri.getPath()
+                    .equals("/login") || relayUri.getPath()
+                    .equals("/login/")) && (relayUri.getQuery() != null && relayUri.getQuery()
+                    .contains("prevurl"))) {
+                relayUri = new URI(relayUri.getScheme(),
+                        relayUri.getUserInfo(),
+                        relayUri.getHost(),
+                        relayUri.getPort(),
+                        relayUri.getQuery()
+                                .replace("prevurl=", ""),
```

Comment:
String `replace` is a noop if it can't match the first parameter. So if replace can't find "prevurl", nothing happens. I think you could delete ` && relayUri.getQuery().contains("prevurl")` from lines 281/282 and it would still work the same.